### DISCUSSION TITLE
Delete a workload item API

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/Microsoft-Fabric-workload-development-sample#123] [General]
Fabric is deleting the workload item from their database, even if the workload backend has not successfully deleted the item. Will this behavior be changed in the near future or Is it going to stay the same?

I felt even the delete endpoint should work similar to the other CRUD operations APIs, where fabric applies the change on those items, only when the workload backend is also able to perform the same operation on the workload.

**Comment by michal-mano** [comment]
Hello,
This is the intended behavior.
If the workload backend does not successfully delete the item, that should not affect the user's experience - they wanted to delete the item and have it no longer appear in their workspace, and so Fabric deletes it regardless of the success of the workload backend.
This is different than other CRUD operations because they *are* dependent on the workload succeeding, whereas an item can be deleted without the workload succeeding.
